src/Avalonia.Controls/Button.cs

```diff
@@ -288,6 +308,11 @@ protected virtual void OnClick()
             }
         }
 
+        protected virtual void OpenFlyout()
+        {
+            _flyout?.ShowAt(this);
```

Comment:
What if flyout was changed while it was opened? Not only on button, but also ContextFlyout one too.
I don't know actually, how it handled in UWP now.

Reply:
Added property changed handler to `Button.Flyout` to handle if the Flyout is changed (close the old flyout first). If the Flyout itself is opened on another control, that logic was already there.